### PR TITLE
[SPARK-40910][SQL] Replace UnsupportedOperationException with SparkUnsupportedOperationException

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This pr aims to replace UnsupportedOperationException with SparkUnsupportedOperationException.

### Why are the changes needed?
1.When I work on https://issues.apache.org/jira/browse/SPARK-40889,
I found `QueryExecutionErrors.unsupportedPartitionTransformError` throw **UnsupportedOperationException**(but not **SparkUnsupportedOperationException**), it seem not to fit into the new error framework.
https://github.com/apache/spark/blob/a27b459be3ca2ad2d50b9d793b939071ca2270e2/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Implicits.scala#L71-L72

2.`QueryExecutionErrors.unsupportedPartitionTransformError` throw SparkUnsupportedOperationException, but UT catch `UnsupportedOperationException`.
https://github.com/apache/spark/blob/a27b459be3ca2ad2d50b9d793b939071ca2270e2/sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala#L288-L301

https://github.com/apache/spark/blob/a27b459be3ca2ad2d50b9d793b939071ca2270e2/sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala#L904-L909

https://github.com/apache/spark/blob/a27b459be3ca2ad2d50b9d793b939071ca2270e2/core/src/main/scala/org/apache/spark/SparkException.scala#L144-L154

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existed UT.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala
Comment (panbingkun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala:301]: image

Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala:301]: In fact, it throws S`parkUnsupportedOperationException`.

**Comment by panbingkun** [comment]
cc @MaxGekk

**Review by itholic — APPROVED** [review]
LGTM when CI pass

File: core/src/main/resources/error/error-classes.json
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:4316]: Let's introduce a appropriate name as you are here.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json]: Done.

File: core/src/main/resources/error/error-classes.json
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:952]: MULTIPLE_BUCKET_TRANSFORM -> MULTIPLE_BUCKET_TRANSFORMS

File: core/src/main/resources/error/error-classes.json
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:954]: Multiple bucket TRANSFORMs

File: core/src/main/resources/error/error-classes.json
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json]: Done

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**
